ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/DirectionsQuestionnaireMapper.java

```diff
@@ -0,0 +1,223 @@
+    private final AddressMapper addressMapper;
+    private final YesNoMapper yesNoMapper;
+
+    @Autowired
+    public DirectionsQuestionnaireMapper(
+        ExpertRowMapper expertRowMapper,
+        UnavailableDateMapper unavailableDateMapper,
+        AddressMapper addressMapper,
+        YesNoMapper yesNoMapper
+    ) {
+        this.expertRowMapper = expertRowMapper;
+        this.unavailableDateMapper = unavailableDateMapper;
+        this.addressMapper = addressMapper;
+        this.yesNoMapper = yesNoMapper;
+    }
+
+    @Override
+    public CCDDirectionsQuestionnaire to(DirectionsQuestionnaire directionsQuestionnaire) {
+        if (directionsQuestionnaire == null) {
+            return null;
+        }
+
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder = CCDDirectionsQuestionnaire.builder();
+        directionsQuestionnaire.getRequireSupport().ifPresent(toRequireSupport(builder));
+
+        toHearingLocation(directionsQuestionnaire.getHearingLocation(), builder);
+
+        directionsQuestionnaire.getWitness().ifPresent(toWitness(builder));
+
+        directionsQuestionnaire.getExpertRequest().ifPresent(toExpertRequest(builder));
+
+        builder.expertReports(directionsQuestionnaire.getExpertReports()
+            .stream()
+            .map(expertRowMapper::to)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList())
+        );
+
+        builder.unavailableDates(directionsQuestionnaire.getUnavailableDates()
+            .stream()
+            .map(unavailableDateMapper::to)
+            .filter(Objects::nonNull)
+            .collect(Collectors.toList())
+        );
+
+        return builder.build();
+    }
+
+    private Consumer<RequireSupport> toRequireSupport(
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        return requireSupport -> {
+            requireSupport.getLanguageInterpreter().ifPresent(builder::languageInterpreted);
+            requireSupport.getSignLanguageInterpreter().ifPresent(builder::signLanguageInterpreted);
+            requireSupport.getOtherSupport().ifPresent(builder::otherSupportRequired);
+
+            requireSupport.getHearingLoop()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::hearingLoop);
+
+            requireSupport.getDisabledAccess()
+                .map(YesNoOption::name)
+                .map(CCDYesNoOption::valueOf)
+                .ifPresent(builder::disabledAccess);
+        };
+    }
+
+    private Consumer<ExpertRequest> toExpertRequest(
+        CCDDirectionsQuestionnaire.CCDDirectionsQuestionnaireBuilder builder
+    ) {
+        return expertRequest -> {
+            builder.expertEvidenceToExamine(expertRequest.getExpertEvidenceToExamine());
+            builder.reasonForExpertAdvice(expertRequest.getReasonForExpertAdvice());
+        };
+    }
+
```

Comment:
Same consumer should not return. https://docs.oracle.com/javase/8/docs/api/java/util/function/Consumer.html.

Reply:
Not sure how to relate that link with your comment.

Reply:
The code is not very readable. Also the Consumer should be defined and inorder to use the builder we are making it a private method and return a consumer to use the builder variable. 

KISS -> "keep it simple, stupid" (design principle)

`directionsQuestionnaire.getExpertRequest().ifPresent(expertRequest -> {
            builder.expertEvidenceToExamine(expertRequest.getExpertEvidenceToExamine());
            builder.reasonForExpertAdvice(expertRequest.getReasonForExpertAdvice());
        });`